internal/region/region.go

```diff
@@ -1,91 +1,133 @@
 package region
 
-// Region represents the sauce labs region.
-type Region uint
-
-const (
-	// None is an undefined sauce labs region.
-	None Region = iota
-	// USWest1 is a sauce labs region in western US, aka us-west-1.
-	USWest1
-	// USEast4 is a sauce labs region in eastern US, aka us-east-4.
-	USEast4
-	// EUCentral1 is a sauce labs region in the EU, aka eu-central-1.
-	EUCentral1
-	// Staging is a sauce labs internal pre-production environment.
-	Staging
+import (
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
 )
 
-var meta = []struct {
-	Name             string
-	APIBaseURL       string
-	AppBaseURL       string
-	WebDriverBaseURL string
-}{
-	// None
+func init() {
+	homeDir, _ := os.UserHomeDir()
+	path := filepath.Join(homeDir, ".sauce", "regions.yml")
+	file, err := os.Open(path)
+	if err != nil {
+		return
+	}
+	defer file.Close()
+
+	if err = yaml.NewDecoder(file).Decode(&userRegionMetas); err != nil {
+		return
+	}
+}
+
+// Region represents the sauce labs region.
+type Region string
+
+type regionMeta struct {
+	Name             string `yaml:"name"`
+	APIBaseURL       string `yaml:"apiBaseURL"`
+	AppBaseURL       string `yaml:"appBaseURL"`
+	WebdriverBaseURL string `yaml:"webdriverBaseURL"`
+}
+
+// None is an undefined sauce labs region.
+const None Region = ""
+
+// USWest1 is a sauce labs region in western US, aka us-west-1.
+const USWest1 Region = "us-west-1"
+
+// USEast4 is a sauce labs region in eastern US, aka us-east-4.
+const USEast4 Region = "us-east-4"
+
+// EUCentral1 is a sauce labs region in the EU, aka eu-central-1.
+const EUCentral1 Region = "eu-central-1"
+
+// Staging is a sauce labs internal pre-production environment.
+const Staging Region = "staging"
+
+var sauceRegionMetas = []regionMeta{
 	{
-		"",
+		None.String(),
 		"",
 		"",
 		"",
 	},
-	// USWest1
 	{
-		"us-west-1",
+		USWest1.String(),
 		"https://api.us-west-1.saucelabs.com",
 		"https://app.saucelabs.com",
 		"https://ondemand.us-west-1.saucelabs.com",
 	},
-	// USEast4
 	{
-		"us-east-4",
+		USEast4.String(),
 		"https://api.us-east-4.saucelabs.com",
 		"https://app.us-east-4.saucelabs.com",
 		"https://ondemand.us-east-4.saucelabs.com",
 	},
-	// EUCentral1
 	{
-		"eu-central-1",
+		EUCentral1.String(),
 		"https://api.eu-central-1.saucelabs.com",
 		"https://app.eu-central-1.saucelabs.com",
 		"https://ondemand.eu-central-1.saucelabs.com",
 	},
-	// Staging
 	{
-		"staging",
+		Staging.String(),
 		"https://api.staging.saucelabs.net",
 		"https://app.staging.saucelabs.net",
 		"https://ondemand.staging.saucelabs.net",
 	},
 }
 
+// userRegionMetas is a list of user defined regions that is loaded
+// from the user's ~/.sauce directory.
+var userRegionMetas = []regionMeta{}
```

Comment:
A bit more canonical:

```suggestion
var userRegionMetas []regionMeta
```